sympy/stats/drv.py

```diff
@@ -177,12 +177,13 @@ def compute_cdf(self, **kwargs):
 
         Returns a Lambda
         """
-        x, z = symbols('x, z', integer=True, cls=Dummy)
+        x = symbols('x', integer=True, cls=Dummy)
+        z = symbols('z', real=True, cls=Dummy)
         left_bound = self.set.inf
 
         # CDF is integral of PDF from left bound to z
         pdf = self.pdf(x)
-        cdf = summation(pdf, (x, left_bound, z), **kwargs)
+        cdf = summation(pdf, (x, left_bound, floor(z)), **kwargs)
```

Comment:
Well, it's been quite some time since I looked at this PR. Can you clarify why we need a `floor` over here and how is it helping?